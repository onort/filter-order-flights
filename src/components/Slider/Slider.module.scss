@import "../../styles/main.scss";

@mixin rangeThumb {
  width: px-to-rem(12px);
  height: px-to-rem(12px);
  border: 0 !important;
  background-color: $primary;
  border-radius: 50%;
  cursor: pointer;
}

.container {
  position: relative;
  width: 100%;
  height: px-to-rem(2px);
  background-color: $grey4;
  outline: none;
}

.slider {
  position: absolute;
  width: 100%;
  height: 0;
  -webkit-appearance: none;

  &::-webkit-slider-thumb {
    -webkit-appearance: none;

    @include rangeThumb;
  }

  &::-moz-range-thumb {
    @include rangeThumb;
  }

  &::-ms-thumb {
    @include rangeThumb;
  }

  &::-webkit-slider-runnable-track {
    height: 10px;
    margin-top: -1px;
    -webkit-appearance: none;
    color: #13bba4;
  }
}
