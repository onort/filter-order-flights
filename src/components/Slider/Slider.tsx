/* stylelint-disable declaration-block-trailing-semicolon */
import React from "react"
import cx from "classnames"

import styles from "./Slider.module.scss"

interface Props {
  className?: string
  currentValue: number
  max: number
  min: number
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
}

const Slider: React.SFC<Props> = props => {
  const { currentValue, max, min, onChange } = props
  const classNames = cx(styles.container, props.className)
  const trackerWidth = (currentValue / max) * 100 - 2
  return (
    <div className={classNames}>
      <div
        className={styles.sliderTrack}
        style={{ width: `${trackerWidth}%` }}
      />
      <input
        type="range"
        min={min}
        max={max}
        value={currentValue}
        className={styles.slider}
        onChange={onChange}
      />
    </div>
  )
}

Slider.defaultProps = {
  currentValue: 0,
  max: 1,
  min: 0
}

export default Slider
