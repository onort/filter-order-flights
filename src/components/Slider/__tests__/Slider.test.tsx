import React from "react"
import { shallow } from "enzyme"

import Slider from "../"

const mockChange = jest.fn()

describe("<Slider />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <Slider onChange={mockChange} max={10} min={0} currentValue={0} />
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should fire onChange function when a change occurs", () => {
    const handleChange = jest.fn()
    const wrapper = shallow(
      <Slider onChange={handleChange} max={10} min={0} currentValue={0} />
    )
    wrapper.find("input").simulate("change", { target: { value: 1 } })
    expect(handleChange).toBeCalledTimes(1)
    expect(handleChange).toBeCalledWith({ target: { value: 1 } })
  })

  it("should be able to have a custom className", () => {
    const testClassName = "testClassName"
    const wrapper = shallow(
      <Slider
        className={testClassName}
        onChange={mockChange}
        max={10}
        min={0}
        currentValue={0}
      />
    )
    expect(wrapper.find("div.container").hasClass(testClassName)).toBe(true)
  })
})
