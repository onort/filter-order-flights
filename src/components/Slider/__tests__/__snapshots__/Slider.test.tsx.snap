// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Slider
    currentValue={0}
    max={10}
    min={0}
    onChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="sliderTrack"
          style={
            Object {
              "width": "-2%",
            }
          }
        />,
        <input
          className="slider"
          max={10}
          min={0}
          onChange={[MockFunction]}
          type="range"
          value={0}
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "sliderTrack",
          "style": Object {
            "width": "-2%",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "slider",
          "max": 10,
          "min": 0,
          "onChange": [MockFunction],
          "type": "range",
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="sliderTrack"
            style={
              Object {
                "width": "-2%",
              }
            }
          />,
          <input
            className="slider"
            max={10}
            min={0}
            onChange={[MockFunction]}
            type="range"
            value={0}
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "sliderTrack",
            "style": Object {
              "width": "-2%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "slider",
            "max": 10,
            "min": 0,
            "onChange": [MockFunction],
            "type": "range",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Slider /> should match snapshot 2`] = `"<div class=\\"container\\"><div class=\\"sliderTrack\\" style=\\"width:-2%\\"></div><input type=\\"range\\" min=\\"0\\" max=\\"10\\" value=\\"0\\" class=\\"slider\\"/></div>"`;
